src/globus_sdk/_remarshal.py

```diff
@@ -0,0 +1,159 @@
+"""
+This module provides internal helpers for remarshalling data from one shape
+into another.
+
+This may be as simple as converting strings from one form to another, or as
+sophisticated as building a specific internal object in a configurable way.
+"""
+
+from __future__ import annotations
+
+import collections.abc
+import typing as t
+import uuid
+
+from globus_sdk._missing import MISSING, MissingType
+from globus_sdk._types import UUIDLike
+
+T = t.TypeVar("T")
+R = t.TypeVar("R")
+
+
+def safe_strseq_iter(
+    value: t.Iterable[t.Any] | str | uuid.UUID,
+) -> t.Iterator[str]:
```

Comment:
Why is UUID the special case for a re-marshaller that will handle any type so long as there are multiple of them?
Would it make more sense to allow an iterable of anything or anything & anything gets converted to a string?

---

Alternatively if it's really meant to be "string or UUID" and not "anything", then we should scope down `t.Iterable` to `t.Iterable[str | UUID]`